app/controllers/SignUpController.scala

```diff
@@ -0,0 +1,64 @@
+package controllers
+
+import scala.concurrent.Future
+
+import com.mohiva.play.silhouette.api._
+import com.mohiva.play.silhouette.api.repositories.AuthInfoRepository
+import com.mohiva.play.silhouette.api.util.PasswordHasherRegistry
+import com.mohiva.play.silhouette.impl.providers.CredentialsProvider
+import javax.inject.Inject
+import play.api.i18n.I18nSupport
+import play.api.mvc.{ AbstractController, AnyContent, ControllerComponents, Request }
+
+import forms.SignUpForm
+import models.services.{ AuthTokenService, UserService }
+import models.{ DatabaseExecutionContext, User}
+import utils.auth.DefaultEnv
+
+class SignUpController @Inject() (
+  components: ControllerComponents,
+  silhouette: Silhouette[DefaultEnv],
+  userService: UserService,
+  passwordHasherRegistry: PasswordHasherRegistry,
+  authInfoRepository: AuthInfoRepository,
+  authTokenService: AuthTokenService,
+  )(implicit ex: DatabaseExecutionContext)
+  extends AbstractController(components) with I18nSupport {
+
+  def view = silhouette.UnsecuredAction.async { implicit request: Request[AnyContent] =>
+    Future.successful(Ok(views.html.signUp(SignUpForm.form)))
+  }
+
+  def submit() = silhouette.UnsecuredAction.async { implicit request: Request[AnyContent] =>
```

Comment:
What's the convention for parens on parameter-less methods that have side effects? `submit` on other controller doesn't have.

Reply:
think should have parens